src/loongarch64/Gresume.c

```diff
@@ -41,31 +41,26 @@ loongarch64_local_resume (unw_addr_space_t as, unw_cursor_t *cursor, void *arg)
     {
       /* Since there are no signals involved here we restore EH and non scratch
          registers only.  */
-      unsigned long sp = uc->uc_mcontext.__gregs[3];
-      unsigned long ra = uc->uc_mcontext.__gregs[1];
-
+      register void *gregs asm("$t0") = uc->uc_mcontext.__gregs;
       asm volatile (
-        "move $t0, %0\n"
-        "move $t1, %1\n"
```

Comment:
The original writting here is trying to avoid the potential registers-conflict but that is not thorough-going.